client/data/agency-dashboard/use-fetch-test-connection.ts

```diff
@@ -1,56 +0,0 @@
-import { useQuery, useQueryClient } from '@tanstack/react-query';
-import { useContext } from 'react';
-import SitesOverviewContext from 'calypso/jetpack-cloud/sections/agency-dashboard/sites-overview/context';
-import { Site } from 'calypso/jetpack-cloud/sections/agency-dashboard/sites-overview/types';
-import wpcom from 'calypso/lib/wp';
-
-const useFetchTestConnection = (
-	isPartnerOAuthTokenLoaded: boolean,
-	isConnectionHealthy: boolean,
-	siteId: number
-) => {
-	const queryClient = useQueryClient();
-	const { search, currentPage, filter, sort } = useContext( SitesOverviewContext );
-	return useQuery( {
-		queryKey: [ 'jetpack-agency-test-connection', siteId ],
-		queryFn: () =>
-			wpcom.req.get(
-				{
-					path: `/jetpack-blogs/${ siteId }/test-connection`,
-					apiNamespace: 'rest/v1.1',
-				},
-				{
-					// We call the current health state "stale", as it might be different than the actual state.
-					is_stale_connection_healthy: Boolean( isConnectionHealthy ),
-				}
-			),
-		enabled: isPartnerOAuthTokenLoaded,
-		refetchOnWindowFocus: false,
```

Comment:
Please note I didn't take `refetchOnWindowFocus` over to the new file. With `staleTime` we're already restricting the frequency of fetches and window focus is a nice way to check whether we're due for a refetch.